_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,693 @@
+
+| Question | Answer |
+|---|---|
+| Type of application? | `Monolithic application` |
+| Name? | `flickr2` |
+| Spring WebFlux? | `No` |
+| Java package name? | `com.auth0.flickr2`  |
+| Type of authentication? | `OAuth 2.0 / OIDC` |
+| Type of database? | `SQL` |
+| Production database? | `PostgreSQL` |
+| Development database? | `H2 with disk-based persistence` |
+| Which cache? | `Ehcache` |
+| Use Hibernate 2nd level cache? | `Yes` |
+| Maven or Gradle? | `Maven` |
+| Use the JHipster Registry? | `No` |
+| Other technologies? | `<blank>` |
+| Client framework? | `React` |
+| Admin UI? | `Yes` |
+| Bootswatch theme? | `United` > `Dark` |
+| Enable i18n? | `Yes` |
+| Native language of application? | `English` |
+| Additional languages? | `Portuguese (Brazilian)` |
+| Additional testing frameworks? | `Cypress` |
+| Install other generators? | `No` |
+
+Press `Enter` and JHipster will create your app in the current directory and run `npm install` to install all the dependencies specified in `package.json`.
+
+### Verify Everything Works with Cypress and Keycloak
+
+When you choose OAuth 2.0 and OIDC for authentication, the users are stored outside of the application, rather than in it. You need to configure an identity provider (IdP) to store your users and allow your app to retrieve information about them. By default, JHipster ships with a Keycloak file for Docker Compose. A default set of users and groups is imported at startup, and it has a client registered for your JHipster app.
+
+Here's what the `keycloak.yml` looks like in your app's `src/main/docker` directory:
+
+```yaml
+# This configuration is intended for development purpose, it's **your** responsibility to harden it for production
+version: '3.8'
+services:
+  keycloak:
+    image: jboss/keycloak:14.0.0
+    command:
+      [
+        '-b',
+        '0.0.0.0',
+        '-Dkeycloak.migration.action=import',
+        '-Dkeycloak.migration.provider=dir',
+        '-Dkeycloak.migration.dir=/opt/jboss/keycloak/realm-config',
+        '-Dkeycloak.migration.strategy=OVERWRITE_EXISTING',
+        '-Djboss.socket.binding.port-offset=1000',
+        '-Dkeycloak.profile.feature.upload_scripts=enabled',
+      ]
+    volumes:
+      - ./realm-config:/opt/jboss/keycloak/realm-config
+    environment:
+      - KEYCLOAK_USER=admin
+      - KEYCLOAK_PASSWORD=admin
+      - DB_VENDOR=h2
+    # If you want to expose these ports outside your dev PC,
+    # remove the "127.0.0.1:" prefix
+    ports:
+      - 127.0.0.1:9080:9080
+      - 127.0.0.1:9443:9443
+      - 127.0.0.1:10990:10990
+```
+
+Start Keycloak with the following command in a terminal window.
+[Install Docker Compose](https://docs.docker.com/compose/install/) if you don't already have it.
+
+```shell
+docker-compose -f src/main/docker/keycloak.yml up
+```
+
+You can verify everything works by starting your app with Maven:
+
+```shell
+./mvnw
+```
+
+Open another terminal to run the Cypress tests:
+
+```shell
+npm run e2e
```

Comment:
I get `npm ERR! Missing script: "e2e"` when I run this command

Reply:
You should get this if you selected Cypress when creating the app.